stdlib/universe/elapsed.go

```diff
@@ -0,0 +1,234 @@
+		return nil, err
+	} else if ok {
+		spec.ColumnName = name
+	} else {
+		spec.ColumnName = "elapsed"
+	}
+
+	return spec, nil
+}
+
+func newElapsedOp() flux.OperationSpec {
+	return new(ElapsedOpSpec)
+}
+
+func (s *ElapsedOpSpec) Kind() flux.OperationKind {
+	return ElapsedKind
+}
+
+type ElapsedProcedureSpec struct {
+	plan.DefaultCost
+	Unit       flux.Duration `json:"unit"`
+	TimeColumn string        `json:"timeColumn"`
+	ColumnName string        `json:"columnName"`
+}
+
+func newElapsedProcedure(qs flux.OperationSpec, pa plan.Administration) (plan.ProcedureSpec, error) {
+	spec, ok := qs.(*ElapsedOpSpec)
+	if !ok {
+		return nil, fmt.Errorf("invalid spec type %T", qs)
+	}
+
+	return &ElapsedProcedureSpec{
+		Unit:       spec.Unit,
+		TimeColumn: spec.TimeColumn,
+		ColumnName: spec.ColumnName,
+	}, nil
+}
+
+func (s *ElapsedProcedureSpec) Kind() plan.ProcedureKind {
+	return ElapsedKind
+}
+
+func (s *ElapsedProcedureSpec) Copy() plan.ProcedureSpec {
+	return &ElapsedProcedureSpec{
+		Unit:       s.Unit,
+		TimeColumn: s.TimeColumn,
+		ColumnName: s.ColumnName,
+	}
+}
+
+func createElapsedTransformation(id execute.DatasetID, mode execute.AccumulationMode, spec plan.ProcedureSpec, a execute.Administration) (execute.Transformation, execute.Dataset, error) {
+	s, ok := spec.(*ElapsedProcedureSpec)
+	if !ok {
+		return nil, nil, fmt.Errorf("invalid spec type %T", spec)
+	}
+	cache := execute.NewTableBuilderCache(a.Allocator())
+	d := execute.NewDataset(id, mode, cache)
+	t := NewElapsedTransformation(d, cache, s)
+	return t, d, nil
+}
+
+type elapsedTransformation struct {
+	d     execute.Dataset
+	cache execute.TableBuilderCache
+
+	unit       float64
+	timeColumn string
+	columnName string
+}
+
+func NewElapsedTransformation(d execute.Dataset, cache execute.TableBuilderCache, spec *ElapsedProcedureSpec) *elapsedTransformation {
+	return &elapsedTransformation{
+		d:     d,
+		cache: cache,
+
+		unit:       float64(spec.Unit),
+		timeColumn: spec.TimeColumn,
+		columnName: spec.ColumnName,
+	}
+}
+
+func (t *elapsedTransformation) RetractTable(id execute.DatasetID, key flux.GroupKey) error {
+	return t.d.RetractTable(key)
+}
+
+func (t *elapsedTransformation) UpdateWatermark(id execute.DatasetID, mark execute.Time) error {
+	return t.d.UpdateWatermark(mark)
+}
+
+func (t *elapsedTransformation) UpdateProcessingTime(id execute.DatasetID, pt execute.Time) error {
+	return t.d.UpdateProcessingTime(pt)
+}
+
+func (t *elapsedTransformation) Finish(id execute.DatasetID, err error) {
+	t.d.Finish(err)
+}
+
+func (t *elapsedTransformation) Process(id execute.DatasetID, tbl flux.Table) error {
+	builder, created := t.cache.TableBuilder(tbl.Key())
+	if !created {
+		return fmt.Errorf("found duplicate table with key: %v", tbl.Key())
+	}
+	cols := tbl.Cols()
+	numCol := 0
+
+	err := execute.AddTableCols(tbl, builder)
+	if err != nil {
+		return err
+	}
+
+	ind := -1
+	for i, c := range cols {
```

Comment:
We have a helper function `execute.ColIdx` that I believe does what you're doing here in this for loop. I think you could get replace this loop entirely with that function here. Just to shrink the size of the code.